demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -16,68 +16,129 @@
+            if height >= max_h or width >= max_w:
+                return
+            self.start_point = (self.start_point[0] - DEFAULT_RESIZE_STEP,
+                                self.start_point[1] - int(DEFAULT_RESIZE_STEP * aspect_ratio))
+            self.end_point = (self.end_point[0] + DEFAULT_RESIZE_STEP,
+                              self.end_point[1] + int(DEFAULT_RESIZE_STEP * aspect_ratio))
+        elif action == 'decrease':
+            if height <= MIN_HEIGHT or width <= MIN_WIDTH:
+                return
+            self.start_point = (self.start_point[0] + DEFAULT_RESIZE_STEP,
+                                self.start_point[1] + int(DEFAULT_RESIZE_STEP * aspect_ratio))
+            self.end_point = (self.end_point[0] - DEFAULT_RESIZE_STEP,
+                              self.end_point[1] - int(DEFAULT_RESIZE_STEP * aspect_ratio))
+        else:
+            raise ValueError("wrong action: {}".format(action))
+
+    def _put_text(self, frame, text):
+        if text == '':
+            return frame
+        text = strip_internal_spaces(text)
+        (txt_w, self._latex_h), baseLine = cv.getTextSize(text, cv.FONT_HERSHEY_SIMPLEX, 1, 3)
+        start_point = (self.start_point[0],
+                       self.end_point[1] - self.start_point[1] + int(self._latex_h * 1.5))
+        frame = cv.putText(frame, text, org=start_point, fontFace=cv.FONT_HERSHEY_SIMPLEX,
+                           fontScale=0.7, color=COLOR_BLACK, thickness=3, lineType=cv.LINE_AA)
+        frame = cv.putText(frame, text, org=start_point, fontFace=cv.FONT_HERSHEY_SIMPLEX,
+                           fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        comment_coords = (0, self.end_point[1] - self.start_point[1] + int(self._latex_h * 1.5))
+        frame = cv.putText(frame, "Predicted:", comment_coords,
+                           fontFace=cv.FONT_HERSHEY_SIMPLEX, fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        return frame
+
+    def put_crop(self, frame, crop):
+        height = self.end_point[1] - self.start_point[1]
+        width = self.end_point[0] - self.start_point[0]
+        crop = cv.resize(crop, (width, height))
+        frame[0:height, self.start_point[0]:self.end_point[0], :] = crop
+        comment_coords = (0, 20)
+        frame = cv.putText(frame, "Model input:", comment_coords, fontFace=cv.FONT_HERSHEY_SIMPLEX,
+                           fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        return frame
+
+    def _put_formula_img(self, frame, formula):
+        if self._renderer is None or formula == '':
+            return frame
+        formula_img = self._render_formula_async(formula)
+        if formula_img is None:
+            return frame
+        y_start = self.end_point[1] - self.start_point[1] + self._latex_h * 2
+        img_shape = formula_img.shape
```

Comment:
```suggestion
        img_shape = formula_img.shape
```
Unused. Delete or use it in lines below.

Reply:
Removed